rabbitmq/tests/test_rabbitmq.py

```diff
@@ -0,0 +1,213 @@
+    aggregator.assert_service_check('rabbitmq.status', status=RabbitMQ.OK)
+
+    aggregator.assert_all_metrics_covered()
+
+
+def test_family_tagging(aggregator, spin_up_rabbitmq, setup_rabbitmq, check):
+    check.check(common.CONFIG_WITH_FAMILY)
+
+    # Node attributes
+    for mname in metrics.COMMON_METRICS:
+        aggregator.assert_metric_has_tag_prefix(mname, 'rabbitmq_node', count=1)
+
+    aggregator.assert_metric('rabbitmq.connections', tags=['rabbitmq_vhost:/'], value=0, count=1)
+    aggregator.assert_metric('rabbitmq.connections', tags=['rabbitmq_vhost:myvhost'], value=0, count=1)
+    aggregator.assert_metric('rabbitmq.connections', tags=['rabbitmq_vhost:myothervhost'], value=0, count=1)
+    for mname in metrics.E_METRICS:
+        aggregator.assert_metric_has_tag('rabbitmq.exchange.%s' %
+                                         mname, 'rabbitmq_exchange_family:test', count=2)
+
+    for mname in metrics.Q_METRICS:
+        aggregator.assert_metric_has_tag('rabbitmq.queue.%s' %
+                                         mname, 'rabbitmq_queue_family:test', count=6)
+
+    aggregator.assert_metric('rabbitmq.connections', tags=['rabbitmq_vhost:/'], value=0, count=1)
+
+    aggregator.assert_service_check('rabbitmq.aliveness', tags=['vhost:/'], status=RabbitMQ.OK)
+    aggregator.assert_service_check('rabbitmq.aliveness', tags=['vhost:myvhost'], status=RabbitMQ.OK)
+    aggregator.assert_service_check('rabbitmq.aliveness', tags=['vhost:myothervhost'], status=RabbitMQ.OK)
+    aggregator.assert_service_check('rabbitmq.status', status=RabbitMQ.OK)
+
+    aggregator.assert_all_metrics_covered()
+
+
+def test_connections(aggregator, spin_up_rabbitmq, setup_rabbitmq, check):
+    # no connections and no 'vhosts' list in the conf don't produce 'connections' metric
+    check.check(common.CONFIG)
+    aggregator.assert_metric('rabbitmq.connections',
+                             tags=['rabbitmq_vhost:/', "tag1:1", "tag2"],
+                             value=0, count=1)
+    aggregator.assert_metric('rabbitmq.connections',
+                             tags=['rabbitmq_vhost:myvhost', "tag1:1", "tag2"],
+                             value=0, count=1)
+    aggregator.assert_metric('rabbitmq.connections',
+                             tags=['rabbitmq_vhost:myothervhost', "tag1:1", "tag2"],
+                             value=0, count=1)
+
+    # no connections with a 'vhosts' list in the conf produce one metrics per vhost
+    aggregator.reset()
+    check.check(common.CONFIG_TEST_VHOSTS)
+    aggregator.assert_metric('rabbitmq.connections', tags=['rabbitmq_vhost:test'], value=0, count=1)
+    aggregator.assert_metric('rabbitmq.connections', tags=['rabbitmq_vhost:test2'], value=0, count=1)
+    aggregator.assert_metric('rabbitmq.connections', count=2)
+
+    with closing(pika.BlockingConnection()), closing(pika.BlockingConnection()):
+        try:
```

Comment:
Can we please get rid of this try? It doesn't seem to be doing anything.

Reply:
done